src/openlcb/MemoryConfigClient.cxxtest

```diff
@@ -391,18 +422,27 @@ TEST_F(MemoryConfigClientTest, writeTimeout)
     expect_any_packet();
     EXPECT_CALL(tSpace_, get_write_timeout())
         .WillRepeatedly(Return(DatagramDefs::TIMEOUT_32));
-    EXPECT_CALL(tSpace_, write(0, _, test_payload.size(), _, _))
+    EXPECT_CALL(tSpace_,
+        write(0, _, test_payload.size(), _, NotifyCapture(&done)))
+        .WillOnce(DoAll(SetArgPointee<3>(ERROR_AGAIN), Return(0)));
+    b = invoke_client_no_block(MemoryConfigClientRequest::WRITE,
+        NodeHandle(TEST_NODE_ID), 0x52, 0, test_payload);
+    wait();
+    clock.advance(SEC_TO_NSEC(15));
+    wait();
+    EXPECT_EQ(MemoryConfigClient::OPERATION_PENDING, b->data()->resultCode);
+    testing::Mock::VerifyAndClearExpectations(&tSpace_);
+    expect_any_packet();
+    EXPECT_CALL(tSpace_,
+        write(0, _, test_payload.size(), _, NotifyCapture(&done)))
         .WillOnce(DoAll(WithArgs<0, 1, 2>(
                 Invoke(this, &MemoryConfigClientTest::AcceptPayload)),
             Return(test_payload.size())));
-    b = invoke_flow(&clientTwo_, MemoryConfigClientRequest::WRITE,
-        NodeHandle(TEST_NODE_ID), 0x52, 0, test_payload);
-    wait();
-    clock.advance(SEC_TO_NSEC(31));
+    done->notify();
```

Comment:
we shouldn't notify done if there is no ERROR_AGAIN in the *error return.

Reply:
nevermind, i mixed up the code. there is an ERROR_AGAIN a while up.
So then I think we should skip the NotifyCapture() in this expectation.